This is the File Management Task

florencekn opened the pull request:
please do check if everything is working on your end, thank you!

florencekn committed:
This is the File Management Task

please do check if everything is working on your end, thank you!

Vacheprime reviewed (changes requested):
Great work. The file is very nicely structured and the comments are very helpful. However, I ran into some issues for some of the commands.

For the fileFileInHomeDir, if you enter a different user from the current one, and you do not have read permission on that user's home folder (for  example, if you enter 'root' as the user), the program will always think that the file does not exist, even if that is not necessarily true. You could try to add a check to see if the home folder is readable, and maybe even use 'sudo' to obtain more privileges. Also, I am not sure if it is needed, but the function does not search for the file recursively.

For the displayTenLargestFiles, on my part, the function displayed 9 directories and 1 file. I do not think that this is the expected output. I believe that it should only check for the size of files.

For the sendFileAsEmailAttachment, in my case it did not work. At first, the 'mutt' command was not found, so I had to install it first, and then I got an error message. Here is the output I got: 
  Enter the username (or enter 'Exit' to leave the program): Danat
  1) Search For a File
  2) Find the 10 Largest Files
  3) Find the 10 Oldest Files
  4) Send File as An Email Attachment
  5) Exit
  Please select an option: 4
  Enter the email address: danatmuradov500@gmail.com
  Enter the file name to send as an email attachment: comments.txt
  Sending the file as an email attachment...
  Error sending message, child exited 127 (Exec error.).
  Could not send the message.
  File sent.

florencekn committed:
Update fileManagement.sh

I made changes on the sendEmailAsAttachment(), made sure to display only files and not directories for the displayTenLargest()

Vacheprime reviewed (approved):
Overall good improvements. The findFileInHomeDir works as expected. The sudo password might not be required everytime so you could add a check to see if the user already has the permissions to access it. Also, you might want to consider listing all of the files found with that name as there could be many.

The displayTenLargestFiles works well, it displays the files as expected. Similarly to the issue we had for findFileInHomeDir, if the user does not have permissions, then it will not display any files. You could add the sudo command (but check if you do have permissions beforehand). The size also seems to be displayed in bytes like this:

5600182272 filepath...
2279866368 filepath...
417816576 filepath...
333237120 filepath...
301794278 filepath...
265993656 filepath...

There are still issues with the send file as attachment, but that seems to be a harder issue to solve.

Approving to start merging code.

florencekn committed:
just removed unnecessary options for find10Oldest() and find10Largest()

florencekn committed:
Added colors to the menu

All the scripts need to be under the same directory for this script to work as well.

florencekn committed:
Removed this file, since I already added the refined menu inside the components folder

florencekn committed:
Merge branch 'florencekn-patch-1' of https://github.com/Vacheprime/SystemMonitor into florencekn-patch-1

florencekn committed:
Added colors to fileManagement, still need to test if the options can have colors

florencekn committed:
finished putting colors

florencekn committed:
added permissions to the script just in case

florencekn committed:
added a return to menu option
